pkg/scheduler/framework/plugins/volumezone/volume_zone.go

```diff
@@ -243,50 +357,138 @@ func (pl *VolumeZone) Filter(ctx context.Context, cs *framework.CycleState, pod
 		}
 		if !ok || !pvTopology.values.Has(v) {
 			logger.V(10).Info("Won't schedule pod onto node due to volume (mismatch on label key)", "pod", klog.KObj(pod), "node", klog.KObj(node), "PV", klog.KRef("", pvTopology.pvName), "PVLabelKey", pvTopology.key)
-			return framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrReasonConflict)
+			return framework.NewStatus(framework.UnschedulableAndUnresolvable, ErrReasonConflict), false
 		}
 	}
-
-	return nil
+	return nil, true
 }
 
-func getStateData(cs *framework.CycleState) (*stateData, error) {
-	state, err := cs.Read(preFilterStateKey)
+// isSchedulableAfterPersistentVolumeClaimChange is invoked whenever a PersistentVolumeClaim added or updated.
+// This function checks whether the PVC makes a pod schedulable.
+func (pl *VolumeZone) isSchedulableAfterPersistentVolumeClaimChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalPVC, modifiedPVC, err := util.As[*v1.PersistentVolumeClaim](oldObj, newObj)
 	if err != nil {
-		return nil, err
+		return framework.QueueAfterBackoff, fmt.Errorf("unexpected objects in isSchedulableAfterPersistentVolumeClaimChange: %w", err)
 	}
-	s, ok := state.(*stateData)
-	if !ok {
-		return nil, errors.New("unable to convert state into stateData")
+	pvModified, isMatched := pl.isSchedulablePersistentVolumeClaim(logger, modifiedPVC, pod)
+	// updated PVC is not schedulable
+	if !isMatched {
+		logger.V(4).Info("PVC was created or updated, but doesn't match the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(modifiedPVC))
+		return framework.QueueSkip, nil
 	}
-	return s, nil
+	pvOriginal, wasMatched := pl.isSchedulablePersistentVolumeClaim(logger, originalPVC, pod)
+	// updated PVC is schedulable and old PVC wasn't schedulable
+	if isMatched && !wasMatched {
+		logger.V(4).Info("PVC was created or updated, and it matches the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(modifiedPVC))
+		return framework.QueueAfterBackoff, nil
+	}
+	// updated PVC is schedulable and PV bound to PVC is changed
+	if pvModified != pvOriginal {
+		logger.V(4).Info("PVC was created or updated, and it matches the pod's PVC and because PV was updated", "pod", klog.KObj(pod), "PVC", klog.KObj(modifiedPVC), "PV", pvModified)
+		return framework.QueueAfterBackoff, nil
+	}
+	logger.V(4).Info("PVC was created or updated, but doesn't match the pod's PVC", "pod", klog.KObj(pod), "PVC", klog.KObj(modifiedPVC))
+	return framework.QueueSkip, nil
 }
 
-func getErrorAsStatus(err error) *framework.Status {
+// isSchedulablePersistentVolumeClaim checks whether pod is schedulable or not.
+func (pl *VolumeZone) isSchedulablePersistentVolumeClaim(logger klog.Logger, pvc *v1.PersistentVolumeClaim, pod *v1.Pod) (string, bool) {
```

Comment:
Rename this function to another. It's unclear what this function does from its name, also the comment is.